### PR TITLE
Add support for Airzone Aidoo Zigbee - AZAI6ZBEMHI

### PR DESCRIPTION
**Pull request by GitCharlieHub**
Added support for Airzone Aidoo Zigbee - AZAI6ZBEMHI

### CODE REVIEW

File: src/devices/AirzoneAidoo.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    toZigbee: [
+        tz.thermostat_local_temperature,
+        tz.thermostat_occupied_heating_setpoint,
+        tz.thermostat_occupied_cooling_setpoint,
+        tz.thermostat_system_mode,
+        tz.on_off,
+        tz.fan_mode,
+    ],
+    exposes: [
+        e.local_temperature(),
```
Comment (Koenkk) [inline src/devices/AirzoneAidoo.ts:25]: Please use `e.climate()` instead (search this repo for examples)

File: src/devices/AirzoneAidoo.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        e.enum('system_mode', ea.ALL, ['off', 'auto', 'cool', 'heat', 'fan_only', 'dry'])
+            .withDescription('Specifies the current operating mode of the thermostat. Supported values are:\n- 0x00 = OFF\n- 0x01 = Auto\n- 0x03 = Cool\n- 0x04 = Heat\n- 0x07 = Fan Only\n- 0x08 = Dry'),
+        e.binary('switch', ea.ALL),
+        e.enum('fan_mode', ea.ALL, ['off', 'low', 'medium', 'high', 'on', 'auto'])
+            .withDescription('Fan mode'),
+    ],
+    endpoint: (device: Device): { [key: string]: number } => {
+        return { 'system': 1 };
+    },
+    meta: { configureKey: 1 } as any,
```
Comment (Koenkk) [inline src/devices/AirzoneAidoo.ts:41]: This can be removed

File: src/devices/AirzoneAidoo.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        e.numeric('occupied_cooling_setpoint', ea.ALL)
+            .withUnit('°C')
+            .withDescription('Specifies the cooling mode setpoint when the room is occupied.'),
+        e.enum('system_mode', ea.ALL, ['off', 'auto', 'cool', 'heat', 'fan_only', 'dry'])
+            .withDescription('Specifies the current operating mode of the thermostat. Supported values are:\n- 0x00 = OFF\n- 0x01 = Auto\n- 0x03 = Cool\n- 0x04 = Heat\n- 0x07 = Fan Only\n- 0x08 = Dry'),
+        e.binary('switch', ea.ALL),
+        e.enum('fan_mode', ea.ALL, ['off', 'low', 'medium', 'high', 'on', 'auto'])
+            .withDescription('Fan mode'),
+    ],
+    endpoint: (device: Device): { [key: string]: number } => {
```
Comment (Koenkk) [inline src/devices/AirzoneAidoo.ts:38]: This can be removed

File: src/devices/AirzoneAidoo.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import {Definition, Device, Endpoint, Logger} from 'zigbee-herdsman-converters';
```
Comment (Koenkk) [inline src/devices/AirzoneAidoo.ts:1]:
```suggestion
import {Definition, Device, Endpoint, Logger} from '../lib/types';
```

File: src/devices/AirzoneAidoo.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import fz from '../converters/fromZigbee';
+import tz from '../converters/toZigbee';
+import * as reporting from '../lib/reporting';
+
+const e = exposes.presets;
+const ea = exposes.access;
+
+const definition: Definition = {
+    zigbeeModel: ['Aidoo Zigbee'],
+    model: 'Aidoo Zigbee AZAI6ZBEMHI',
```
Comment (Koenkk) [inline src/devices/AirzoneAidoo.ts:12]:
```suggestion
    model: 'AZAI6ZBEMHI',
```

File: src/devices/AirzoneAidoo.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        e.local_temperature(),
+        e.numeric('occupied_heating_setpoint', ea.ALL)
+            .withUnit('°C')
+            .withDescription('Occupied heating setpoint'),
+        e.numeric('occupied_cooling_setpoint', ea.ALL)
+            .withUnit('°C')
+            .withDescription('Specifies the cooling mode setpoint when the room is occupied.'),
+        e.enum('system_mode', ea.ALL, ['off', 'auto', 'cool', 'heat', 'fan_only', 'dry'])
+            .withDescription('Specifies the current operating mode of the thermostat. Supported values are:\n- 0x00 = OFF\n- 0x01 = Auto\n- 0x03 = Cool\n- 0x04 = Heat\n- 0x07 = Fan Only\n- 0x08 = Dry'),
+        e.binary('switch', ea.ALL),
```
Comment (Koenkk) [inline src/devices/AirzoneAidoo.ts:34]:
```suggestion
        e.switch(),
```

File: src/devices/AirzoneAidoo.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        e.enum('fan_mode', ea.ALL, ['off', 'low', 'medium', 'high', 'on', 'auto'])
```
Comment (Koenkk) [inline src/devices/AirzoneAidoo.ts:35]:
```suggestion
        e.fan().withModes(['off', 'low', 'medium', 'high', 'on', 'auto'])
```

File: src/devices/AirzoneAidoo.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    configure: async (device: Device, coordinatorEndpoint: Endpoint, logger: Logger): Promise<void> => {
+        const endpoint = device.getEndpoint(1);
+        await reporting.bind(endpoint, coordinatorEndpoint, ['hvacThermostat', 'hvacFanCtrl', 'genOnOff'], logger);
+        await reporting.thermostatTemperature(endpoint, logger);
+        await reporting.thermostatOccupiedHeatingSetpoint(endpoint, logger);
+        await reporting.thermostatOccupiedCoolingSetpoint(endpoint, logger);
+        await reporting.onOff(endpoint, logger);
+        await reporting.fanMode(endpoint, logger);
+    },
+    device: {
```
Comment (Koenkk) [inline src/devices/AirzoneAidoo.ts:51]: This is not used, can be removed

File: src/devices/AirzoneAidoo.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+            e.climate()
+                .withLocalTemperature()
+                .withSystemMode(['off', 'auto', 'cool', 'heat', 'fan_only', 'dry'])
+                .withRunningState(['idle', 'heat', 'cool'])
+                .withFanMode(['off', 'low', 'medium', 'high', 'on', 'auto'])
+                .withSetpoint('occupied_heating_setpoint', 5, 30, 0.5)
+                .withSetpoint('occupied_cooling_setpoint', 5, 30, 0.5)
+        ]
+    },
+    ha_category: 'climate'
```
Comment (Koenkk) [inline src/devices/AirzoneAidoo.ts:63]: This is not used, can be removed

**Comment by GitCharlieHub** [comment]
@Koenkk thanks for there pointers. used e.climate as you pointed out.

Still getting the following build error and clueless as on how to fix:
Error: src/devices/AirzoneAidoo.ts(1,21): error TS2305: Module '"../lib/types"' has no exported member 'Device'.
Error: src/devices/AirzoneAidoo.ts(1,29): error TS2305: Module '"../lib/types"' has no exported member 'Endpoint'.
Error: src/devices/AirzoneAidoo.ts(33,16): error TS7006: Parameter 'device' implicitly has an 'any' type.
Error: src/devices/AirzoneAidoo.ts(37,23): error TS7006: Parameter 'device' implicitly has an 'any' type.
Error: src/devices/AirzoneAidoo.ts(37,31): error TS7006: Parameter 'coordinatorEndpoint' implicitly has an 'any' type.
Error: src/devices/AirzoneAidoo.ts(37,52): error TS7006: Parameter 'logger' implicitly has an 'any' type.

I did the swap you mentioned above:
```
- import {Definition, Device, Endpoint, Logger} from 'zigbee-herdsman-converters';
+ import {Definition, Device, Endpoint, Logger} from '../lib/types';
```

**Comment by AlexisPolegato** [comment]
File name shouldn't be "airzone.ts" ?
Airzone is the manufacturer name and Aidoo is the device name.

Regarding the Model / Desciption : The device can be used to control remotely an AC, it is compatible with many different kinds of AC Brands. Here you mention only the version compatible with Mitshubishi.
=> https://myzone.airzonefrance.fr/airzone-products/current-product-lines/aidoo/zigbee/?limit=15&page=1

In terms of Zigbee functionnality, it is only 1 product, so I think it would make more sense to make something generic as done in #7405 otherwise we would need to create dozens of reference.

Also, the ACLouverPosition Attribute is missing here.

**Comment by Koenkk** [comment]
@AlexisPolegato could you submit a pr to fix these problems?

**Pull request merged**